torch/distributed/_sharded_tensor/ops/elementwise_ops.py

```diff
@@ -0,0 +1,35 @@
+import copy
+
+import torch
+from torch.distributed._sharded_tensor import (
+    sharded_op_impl,
+    Shard,
+    ShardedTensor,
+)
+
+
+@sharded_op_impl(torch.nn.functional.gelu)
+def gelu(types, args=(), kwargs=None, pg=None):
+    """
+    Handles ``__torch_function__`` dispatch for the elementwise op
+    ``torch.nn.functional.gelu``.
+    This method computes on either a normal tensor or a sharded tensor.
+    """
+    input = args[0]
+    op = torch.nn.functional.gelu
+    # Validate types
+    if not isinstance(input, torch.Tensor) and not isinstance(input, ShardedTensor):
+        raise TypeError("input needs to be either torch.Tensor or ShardedTensor")
+    if isinstance(input, torch.Tensor):
+        return op(args[0])
```

Comment:
Do we need to handle this case at all? Shouldn't normal tensor dispatch take care of this? My understanding is that this codepath will never be executed given how `__torch_function__` dispatch works.

Reply:
I see.. That makes sense. Thanks for pointing that out.